e2e/app/component/data-resource-card.ts

```diff
@@ -47,23 +47,20 @@ export default class DataResourceCard extends CardBase {
     return fp.shuffle(cards)[0];
   }
 
-  static async findCard(page: Page, resourceName: string, timeOut = 60000): Promise<DataResourceCard | null> {
-    const selector = `.//*[${DataResourceCardSelector.cardNameXpath} and normalize-space(text())="${resourceName}"]`;
-    try {
-      await page.waitForXPath(selector, { visible: true, timeout: timeOut });
-    } catch (err) {
-      return null;
-    }
-    const allCards = await this.findAllCards(page);
-    for (const card of allCards) {
-      const handle = card.asElementHandle();
-      const children = await handle.$x(selector);
-      if (children.length > 0) {
-        return card; // matched resource name.
-      }
-      await handle.dispose(); // not it, dispose the ElementHandle.
-    }
-    return null; // not found
+  static async findCard(page: Page, resourceName: string, timeout = 30000): Promise<DataResourceCard | null> {
+    const selector =
+      `${DataResourceCardSelector.cardRootXpath}[.//*[${DataResourceCardSelector.cardNameXpath}` +
+      ` and normalize-space(text())="${resourceName}"]]`;
+    return page
+      .waitForXPath(selector, { timeout })
+      .then((element) => {
+        logger.info(`Found data resource card: "${resourceName}"`);
```

Comment:
How often do we use the log information for the these scenarios ?

Reply:
I use log for test failure investigation often. It's verbose but useful. Thank you very much.